espresso/streamer_test.go

```diff
@@ -0,0 +1,610 @@
+package espresso_test
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"log/slog"
+	"math/big"
+	"math/rand"
+	"testing"
+	"time"
+
+	esp_client "github.com/EspressoSystems/espresso-network-go/client"
+	esp_common "github.com/EspressoSystems/espresso-network-go/types"
+	"github.com/ethereum-optimism/optimism/espresso"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/crypto"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	opsigner "github.com/ethereum-optimism/optimism/op-service/signer"
+	"github.com/ethereum-optimism/optimism/op-service/testutils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	geth_types "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// TestNewEspressoStreamer tests that we can create a new EspressoStreamer
+// without any errors being thrown.
+
+func TestNewEspressoStreamer(t *testing.T) {
+	defer func() {
```

Comment:
Is this somehow different from just writing `_, err := NewEspressoStreamer()` and checking the error?

Reply:
It is in a way.  Generally speaking, the call to `NewEspressoStreamer` doesn't have an explicit error case where it able to return an error.  In treating `NewEspressoStreamer` as a black box, we we that it cannot return an error explicitly due to the signature, but there are no guarantees that it's call won't `panic`.

The deferred call to `recover` will catch all panics that occur, and allow us to report them. Now, in inspecting the code currently, this doesn't seem like it could ever happen.  But, having this check here, prevents us from introducing such a "feature" in the future.

Reply:
Though, to be fair.  I imagine that `panic` would generate an automatic fail anyway, so this is likely unnecessary entirely.

Reply:
Addressed in d60fdec94fd542878bb99c87e1b962e2e3ab95b0.